packages/headless/src/controllers/query-summary/headless-query-summary.ts

```diff
@@ -1,21 +1,82 @@
 import {Engine} from '../../app/headless-engine';
 import {PaginationSection, SearchSection} from '../../state/state-sections';
-import {buildController} from '../controller/headless-controller';
+import {buildController, Controller} from '../controller/headless-controller';
 import {buildSearchStatus} from '../search-status/headless-search-status';
 
-export type QuerySummary = ReturnType<typeof buildQuerySummary>;
-/** The state relevant to the `QuerySummary` controller.*/
-export type QuerySummaryState = QuerySummary['state'];
-
 /**
- * The QuerySummary component can display information about the current range of results (e.g., "Results
+ * The `QuerySummary` controller provides information about the current query and results (e.g., "Results
  * 1-10 of 123").
+ * */
+export interface QuerySummary extends Controller {
+  /** The state relevant to the `QuerySummary` controller.*/
+  state: QuerySummaryState;
+}
+
+export interface QuerySummaryState {
```

Comment:
could it extend the SearchStatusState instead of re-documenting its properties?

Reply:
I think it could be possible with an adjustment to the parser. Good suggestion, I will explore,